test/stablecoin_converter.js

```diff
@@ -640,7 +651,37 @@ contract("StablecoinConverter", async (accounts) => {
         stablecoinConverter.submitSolution(batchIndex, owner, orderId, volume, prices, tokenIdForPrice)
       )
     })
-    it("reverts, if a trade touches a token, for which no price is provided", async () => {
+    it("reverts, if a trade touches a token, for which no price is provided and hence the binary search for the price fails", async () => {
```

Comment:
I still don't understand how in one case the binary search fails and in the other one we divide by 0. I'd expect a trade for which no price is provided to behave the same.

Reply:
In the one test the price is missing in the buyToken and the other one, the price is missing in the sellToken. 
In case of the missing in the sellToken, we have to protect ourselves first from dividing by 0. The binary search would, of course, fail as well, but only later. 
Instead, if the buyToken price is missing, we would first try to search for a price with the binary search and run into issues there.

Reply:
Would be awesome to call it

`reverts, if price for sellToken not specified`

and

`reverts, if price for buyToken not specified`